src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanUtils.java

```diff
@@ -55,4 +58,43 @@ public void visit( final LogicalOpUnion op ) {
 		}
 	} // end of class SourceAssignmentChecker
 
+	public static void printStringOfFm( final StringBuilder builder, final FederationMember fm ) {
+
+		final DataRetrievalInterface intFace = fm.getInterface();
+		if ( intFace instanceof SPARQLEndpointInterface){
+			builder.append( ((SPARQLEndpointInterface) intFace).getURL() );
+		}
+		else {
+			builder.append( " Print the Federation Member in the type of interface: " + intFace.getClass().getName() + "is an open TODO" );
+		}
```

Comment:
In addition to the URL, the string that is produced here should also contain an indication of the type of fed.member/interface. For instance, possible example strings could be:

"SPARQL endpoint at http://dbpedia.org/sparql"

"TPF server at TODO"